// @node-redis/json/dist/commands/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.transformNumbersReply = exports.transformRedisJsonNullReply = exports.transformRedisJsonReply = exports.transformRedisJsonArgument = void 0;
const ARRAPPEND = require("@node-redis/json/dist/commands/ARRAPPEND");
const ARRINDEX = require("@node-redis/json/dist/commands/ARRINDEX");
const ARRINSERT = require("@node-redis/json/dist/commands/ARRINSERT");
const ARRLEN = require("@node-redis/json/dist/commands/ARRLEN");
const ARRPOP = require("@node-redis/json/dist/commands/ARRPOP");
const ARRTRIM = require("@node-redis/json/dist/commands/ARRTRIM");
const DEBUG_MEMORY = require("@node-redis/json/dist/commands/DEBUG_MEMORY");
const DEL = require("@node-redis/json/dist/commands/DEL");
const FORGET = require("@node-redis/json/dist/commands/FORGET");
const GET = require("@node-redis/json/dist/commands/GET");
const MGET = require("@node-redis/json/dist/commands/MGET");
const NUMINCRBY = require("@node-redis/json/dist/commands/NUMINCRBY");
const NUMMULTBY = require("@node-redis/json/dist/commands/NUMMULTBY");
const OBJKEYS = require("@node-redis/json/dist/commands/OBJKEYS");
const OBJLEN = require("@node-redis/json/dist/commands/OBJLEN");
const RESP = require("@node-redis/json/dist/commands/RESP");
const SET = require("@node-redis/json/dist/commands/SET");
const STRAPPEND = require("@node-redis/json/dist/commands/STRAPPEND");
const STRLEN = require("@node-redis/json/dist/commands/STRLEN");
const TYPE = require("@node-redis/json/dist/commands/TYPE");
exports.default = {
    ARRAPPEND,
    arrAppend: ARRAPPEND,
    ARRINDEX,
    arrIndex: ARRINDEX,
    ARRINSERT,
    arrInsert: ARRINSERT,
    ARRLEN,
    arrLen: ARRLEN,
    ARRPOP,
    arrPop: ARRPOP,
    ARRTRIM,
    arrTrim: ARRTRIM,
    DEBUG_MEMORY,
    debugMemory: DEBUG_MEMORY,
    DEL,
    del: DEL,
    FORGET,
    forget: FORGET,
    GET,
    get: GET,
    MGET,
    mGet: MGET,
    NUMINCRBY,
    numIncrBy: NUMINCRBY,
    NUMMULTBY,
    numMultBy: NUMMULTBY,
    OBJKEYS,
    objKeys: OBJKEYS,
    OBJLEN,
    objLen: OBJLEN,
    RESP,
    resp: RESP,
    SET,
    set: SET,
    STRAPPEND,
    strAppend: STRAPPEND,
    STRLEN,
    strLen: STRLEN,
    TYPE,
    type: TYPE
};
function transformRedisJsonArgument(json) {
    return JSON.stringify(json);
}
exports.transformRedisJsonArgument = transformRedisJsonArgument;
function transformRedisJsonReply(json) {
    return JSON.parse(json);
}
exports.transformRedisJsonReply = transformRedisJsonReply;
function transformRedisJsonNullReply(json) {
    if (json === null)
        return null;
    return transformRedisJsonReply(json);
}
exports.transformRedisJsonNullReply = transformRedisJsonNullReply;
function transformNumbersReply(reply) {
    return JSON.parse(reply);
}
exports.transformNumbersReply = transformNumbersReply;
