// @node-redis/bloom/dist/commands/cuckoo/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.pushInsertOptions = void 0;
const ADD = require("@node-redis/bloom/dist/commands/cuckoo/ADD");
const ADDNX = require("@node-redis/bloom/dist/commands/cuckoo/ADDNX");
const COUNT = require("@node-redis/bloom/dist/commands/cuckoo/COUNT");
const DEL = require("@node-redis/bloom/dist/commands/cuckoo/DEL");
const EXISTS = require("@node-redis/bloom/dist/commands/cuckoo/EXISTS");
const INFO = require("@node-redis/bloom/dist/commands/cuckoo/INFO");
const INSERT = require("@node-redis/bloom/dist/commands/cuckoo/INSERT");
const INSERTNX = require("@node-redis/bloom/dist/commands/cuckoo/INSERTNX");
const LOADCHUNK = require("@node-redis/bloom/dist/commands/cuckoo/LOADCHUNK");
const RESERVE = require("@node-redis/bloom/dist/commands/cuckoo/RESERVE");
const SCANDUMP = require("@node-redis/bloom/dist/commands/cuckoo/SCANDUMP");
const generic_transformers_1 = require("@node-redis/client/dist/lib/commands/generic-transformers");
exports.default = {
    ADD,
    add: ADD,
    ADDNX,
    addNX: ADDNX,
    COUNT,
    count: COUNT,
    DEL,
    del: DEL,
    EXISTS,
    exists: EXISTS,
    INFO,
    info: INFO,
    INSERT,
    insert: INSERT,
    INSERTNX,
    insertNX: INSERTNX,
    LOADCHUNK,
    loadChunk: LOADCHUNK,
    RESERVE,
    reserve: RESERVE,
    SCANDUMP,
    scanDump: SCANDUMP
};
function pushInsertOptions(args, items, options) {
    if (options === null || options === void 0 ? void 0 : options.CAPACITY) {
        args.push('CAPACITY');
        args.push(options.CAPACITY.toString());
    }
    if (options === null || options === void 0 ? void 0 : options.NOCREATE) {
        args.push('NOCREATE');
    }
    args.push('ITEMS');
    return (0, generic_transformers_1.pushVerdictArguments)(args, items);
}
exports.pushInsertOptions = pushInsertOptions;
